discovery-infra/tests/base_test.py

```diff
@@ -77,3 +82,10 @@ def assert_cluster_validation(cluster_info, validation_section, validation_id, e
     def assert_string_length(string, expected_len):
         assert len(string) == expected_len, "Expected len string of: " + str(expected_len) + \
                                             " rather than: " + str(len(string)) + " String value: " + string
+
+    @staticmethod
+    def collect_test_logs(cluster, api_client, test_name):
+        log_dir_name = f'{consts.LOG_FOLDER}/{test_name}'
+        with suppress(ApiException):
+            cluster_detials = json.loads(json.dumps(cluster.get_details().to_dict(), sort_keys=True, default=str))
+            download_logs(api_client, cluster_detials, log_dir_name, True)
```

Comment:
Be aware that downloading `must-gather` would explode your storage. I think you should consider if you want to download it

Reply:
These logs are helpful for debugging. I can  download must-gather in the case failure. wdty?

Reply:
I think it should be done.
You can keep it for now, and look if it's too much.

Reply:
changed to only download must-gather in the case of test failure